microhapulator/cli/pipe.py

```diff
@@ -43,19 +45,28 @@ def main(args):
         thresh_gap_alert=args.gap_alert,
         hspace=args.hspace,
     )
-    snakefile = files("microhapulator") / "workflows" / "analysis.smk"
-    success = snakemake(
-        snakefile,
-        cores=args.threads,
-        printshellcmds=True,
-        dryrun=args.dryrun,
-        config=config,
-        workdir=args.workdir,
-    )
-    if not success:
+    try:
+        run_snakemake(config, args.workdir, cores=args.threads, dryrun=args.dryrun)
+        return 0
+    except WorkflowError:
         return 1
 
 
+def run_snakemake(config, workdir, cores=1, dryrun=False):
+    outcfg = smk_types.OutputSettings(printshellcmds=True)
+    with smk_api(outcfg) as smk:
+        workflow = smk.workflow(
+            config_settings=smk_types.ConfigSettings(config=config),
+            storage_settings=smk_types.StorageSettings(),
+            resource_settings=smk_types.ResourceSettings(cores=cores),
+            snakefile=files("microhapulator") / "workflows" / "analysis.smk",
+            workdir=Path(workdir),
+        )
+        dag = workflow.dag(smk_types.DAGSettings(targets=["report"]))
+        mode = "dryrun" if dryrun else "local"
+        dag.execute_workflow(executor=mode)
```

Comment:
This is close to the minimal code required to replicate the flags I commonly used with the `snakemake.snakemake` command in version 7 and below.